### ISSUE TITLE
Help wanted using hackstrings provided (e.g. 991cnx_read_rom_checksum/segment0.asm)

### ISSUE THREAD

**Issue by 2t22Ydhr** [user202729/CasioEmu#4]
Dear user, 

Could I please have some instructions about how to enter the hackstrings you have the ability to compile on a fx991ex calculator? I want to try and dump my ROM. I see you have posted instructions at https://tieba.baidu.com/p/6706121713?pn=3 for the CNX, but I am unsure what I need to do beforehand to enter these instructions, and don't know if they apply to the fx991ex as well. 

I am doing this ultimately because as of now I am trying to enter 100+50an mode on the 580VNX emulator (real ROM). I know this works on my 991ex, but I am not sure if this works on the 580VNX (I don't have one). If it does, is there some reason it is not possible to do this on model/fx580vnx? 

I am very sorry if this is a waste of your time. Any help would be appreciated.

**Comment by user202729** [comment]
Programming in ROP/discovering locations of gadgets is very time-consuming and currently I don't really have time/want to do that. If you try to learn the techniques yourself and have some time maybe I can give some hints, but not much.

Not sure about the difference between simulator and real calculator, but usually the simulator behavior should be identical to the real one. And yes, usually techniques on one machine does not really work on others unless they're something "generic" such as the basic overflow etc.

**Comment by 2t22Ydhr** [comment]
Yep. It's very time consuming (and I don't have extreme amounts of time either with A-levels / STEP examinations). 

I was just wondering about why 100+50an method didn't work on the calculator simulator (e.g. if there was an issue in the emulator / the ROM wasn't the exact same). 

This is what I am doing: 
![image](https://user-images.githubusercontent.com/125202471/218470644-8017d98f-fddf-4bea-af7b-f16a43630b61.png)

I am only asking if you have heard of this issue or tried 100+50an on the emulator before: it would be far too time consuming to do anything else. 

This is what happens when I try doing 100+50an: 

https://user-images.githubusercontent.com/125202471/218472903-55e7e183-347f-4356-87e4-7a3b553c7bab.mp4



Here is also a video of the memory: 
https://user-images.githubusercontent.com/125202471/218472173-275d4977-b23d-4ccf-9f89-c929fb363a89.mp4


I will consult around to see if there is another method of doing what I want on the emulator.

**Comment by user202729** [comment]
I don't recall having seen something similar before...

A blind guess, the real hardware might treat segment 0x0F the same as 0x03...

I'm not quite sure what you're trying to do exactly, but usually if I want to execute some hack in the emulator I just directly patch the memory to force the input size to be 200 bytes (use the `inject` Lua function), then click [=] (to the best of my recollection), so there's no need to enter auxiliary modes on the emulator.

**Comment by user202729** [comment]
Right, I forgot what the an does...

if I recalled correctly, it sets the stack pointer to somewhere in the input (?)

then you would actually need to enter something meaningful for it to work?

**Comment by erik-404err** [comment]
Dear @2t22Ydhr 
i know it has been some time till you worked on this. I would love to get guidance form you on how to get this project to emulate a fx991ex. I have extracted my own ROM for it, but it does not run. Yours seemed do run well. could you provide an md5 hash of your ROM file and maybe your ./models folder?
would be greatly appreciated.
